### DISCUSSION TITLE
TabbedBar: BarItemType: Enabled and Label properties

### DISCUSSION THREAD

**Discussion by tyroneBaba** [appcelerator/titanium_mobile#12664] [Q&A] [answered]
https://docs.appcelerator.com/platform/latest/#!/api/Titanium.UI.TabbedBar

I'm trying to set the BarItemType to disabled=true (its enabled by default) to disable the tab but it doesn't seem to do anything on Android, it is fine on iOS. Not sure if I'm missing something or doing something wrong.

Also, is it possible to change the text colour and/or the selected colour on the labels on Android or iOS

`
			<TabbedBar id='pageTab'  width="Ti.UI.SIZE" height='Ti.UI.SIZE' index='0' bottom='5' onClick='onRefreshStart' borderRadius='10' backgroundColor='#80000000'>
				<BarItemTypes>
					<BarItemType>Test1</BarItemType>
					<BarItemType enabled='false'>Test2</BarItemType>
				</BarItemTypes>
			</TabbedBar>
`

**Comment by m1ga** [comment, marked as answer]
**Android**

For the colors on Android you have to create a theme:
```
	<style name="Theme.MyTheme" parent="@style/Theme.Titanium"></style>

	<style name="Base.Widget.Design.TabLayout" parent="android:Widget">
	    <item name="tabIndicatorColor">#0f0</item>
	    <item name="tabSelectedTextColor">#00f</item>
	    <item name="tabIndicatorHeight">4dp</item>
	</style>
```

![Screenshot_20210326-132442](https://user-images.githubusercontent.com/4334997/112631069-b7ebe380-8e36-11eb-9017-03c0f6945fdc.png)

The parent is plain widget thats why the padding and everything it missing. Either find the correct parent or add the padding again :smile:

**Reply by jquick-axway** [reply]
Correct. The "enable" property is ignored on Android. Mostly because it's not natively supported on Android for the default "top" tab layout style. At least as far as I can see. We can make it not clickable, but it won't show that disabled visual state.

That said, if you're using `Ti.UI.TabbedBar` to show a bar of options instead of implementing your own tab group like interface, then you'll want to use [Ti.UI.OptionBar](https://docs.appcelerator.com/platform/latest/#!/api/Titanium.UI.OptionBar) which will be available in the upcoming Titanium 10.0.0 version. It will show a toggle-able button bar on Android and use a segmented control on iOS. You can see screenshots of it below.
https://github.com/appcelerator/titanium_mobile/pull/12510

**Reply by tyroneBaba** [reply]
Thank you both

**Reply by Max87ZA** [reply]
Don't forget to apply that tablayout style inside your theme:
`<item name="tabStyle">@style/Base.Widget.Design.TabLayout</item> `
And one important thing(took me a while to get it) - if you're using night mode, don't forget to edit /res/values-night.xml also...
